snare/middlewares.py

```diff
@@ -17,10 +16,15 @@ def __init__(self, error_404, error_500=None, headers=[], server_header=""):
             self.headers["Server"] = server_header
 
     async def handle_404(self, request):
-        return aiohttp_jinja2.render_template(self.error_404, request, {})
-
-    async def handle_500(self, request):
-        return aiohttp_jinja2.render_template(self.error_500, request, {})
+        if not self.error_404:
```

Comment:
@lordlabuckdas Can you please explain what is happening here? Why are we triggering NotFound error when `self.error_404` is not set? Shouldn't it be triggered when `self.error_404` is set?

Reply:
`self.error_404` contains the hash/file name of the 404 page. In case websites do not have a 404 template set, `self.error_404` should be `None` and so we're triggering the exception manually with the headers. If the file is present, we're serving the page using Jinja's `render_template` function.

Reply:
Oh okay, I thought a boolean value would be assigned to the error_404 variable.

Your explanation makes sense.